source/base/Settings.cpp

```diff
@@ -459,4 +460,30 @@ void Settings::setResourceId(QString resourceId)
     QSettings settings;
     settings.setValue("authentication/resourceId", resourceId);
     emit resourceIdChanged(resourceId);
-}
\ No newline at end of file
+}
+
+QDateTime Settings::getLatestMamSyncDate() const
+{
+    QDateTime returnValue = QDateTime::currentDateTimeUtc().addDays(-15);
+    QSettings settings;
+    QString jid=getJid();
```

Comment:
This must be the jid of the other message partner or the group jid. As a result, there will be one entry in the settings file for each contact.

setLatestMamSyncDate is called after a Mam IQ Fin message is received. On the next request of the next group to catch MaM messages, the start-time is 'now'. If there are multiple groups, only the first group is requested with the correct last date. The next groups are requested with a start date of 'now'.